kong/runloop/mesh.lua

```diff
@@ -139,7 +147,11 @@ local function init()
     -- We (ab)use the OpenSSL hostname callback for doing this
     kong.default_client_ssl_ctx:setHostNameCallback(client_hostname_callback, mesh_client_ssl_ctx, mesh_alpn)
     if ngx.config.subsystem == "http" then
-      kong.default_client_ssl_ctx:setAlpnProtos { mesh_alpn, "http/1.1" }
+      if http2_enabled then
+        kong.default_client_ssl_ctx:setAlpnProtos { mesh_alpn, "h2", "http/1.1" }
```

Comment:
@james-callahan should this change be removed?

Reply:
no? I think it looks okay.

Need to test with multiple proxys (one with http2, one without) going to different upstreams (one supporting http2, one not)

Reply:
proxy module does not support http2, afaik.

Reply:
ah yes indeed. as nginx doesn't support http2 to the upstream then we shouldn't advertise support here.

Reply:
Removed that.